src/api/libcellml/generatorprofile.h

```diff
@@ -3018,6 +3017,217 @@ class LIBCELLML_EXPORT GeneratorProfile
     void setExternalVariableMethodCallString(bool forDifferentialModel,
                                              const std::string &externalVariableMethodCallString);
 
+    /**
+     * @brief Get the @c std::string for the data structure for the root finding
+     * information object.
+     *
+     * Return the @c std::string for the data structure for the root finding
+     * information object.
+     *
+     * @param forDifferentialModel Whether the call to the root finding
+     * information object is for a differential model, as opposed to an
+     * algebraic model.
+     *
+     * @return The @c std::string for the data structure for the root finding
+     * information object.
+     */
+    std::string rootFindingInfoObjectString(bool forDifferentialModel) const;
+
+    /**
+     * @brief Set the @c std::string for the data structure for the root finding
+     * information object.
+     *
+     * Set the @c std::string for the data structure for the root finding
+     * information object.
+     *
+     * @param forDifferentialModel Whether the call to the root finding
+     * information object is for a differential model, as opposed to an
+     * algebraic model.
+     * @param rootFindingInfoObjectString The @c std::string to use for the data
+     * structure for the root finding information object.
+     */
+    void setRootFindingInfoObjectString(bool forDifferentialModel,
+                                        const std::string &rootFindingInfoObjectString);
+
+    /**
+     * @brief Get the @c std::string for the extern NLA solve method.
```

Comment:
should this be `external` rather than `extern` ?

Reply:
No, it is correct. This method is used to specify the definition of the method that is used to call the NLA solver, i.e.
```c
extern void nlaSolve(void (*objectiveFunction)(double *, double *, void *),
                     double *u, int n, void *data);
```

Reply:
`extern` is a C keyword.  `external` is the English word that that should be used here.  I can't see how a C keyword is the correct choice.

Reply:
92405cf2